pkg/user/tx_client.go

```diff
@@ -445,26 +441,26 @@ func (client *TxClient) ConfirmTx(ctx context.Context, txHash string) (*TxRespon
 
 		if err == nil && resp != nil {
 			switch resp.Status {
-			// FIXME: replace hardcoded status with constants
-			case "PENDING":
+			case core.TxStatusPending:
 				// Continue polling if the transaction is still pending
 				select {
 				case <-ctx.Done():
 					return &TxResponse{}, ctx.Err()
 				case <-pollTicker.C:
 					continue
 				}
-			case "COMMITTED":
+			case core.TxStatusCommitted:
 				txResponse := &TxResponse{
 					Height: resp.Height,
 					TxHash: txHash,
 					Code:   resp.ExecutionCode,
+					Error:  resp.Error,
 				}
 				if resp.ExecutionCode != 0 {
-					return txResponse, fmt.Errorf("tx was included but failed with code %d: %s", resp.ExecutionCode, resp.Status)
+					return txResponse, fmt.Errorf("tx was committed but failed with code %d: %s", resp.ExecutionCode, resp.Status)
```

Comment:
This could change to include the error
```suggestion
					return txResponse, fmt.Errorf("tx was committed but failed with code %d: %s", resp.ExecutionCode, resp.Error)
```